### PR TITLE
open_anndata unit test

### PR DESCRIPTION
**Pull request by ebezzi**
Unit test for the `open_anndata` function.

The way I implemented fixtures for this test isn't very elegant but it's functional. Suggestions on how to improve it are welcome.

### CODE REVIEW

File: tools/cell_census_builder/tests/anndata/test_anndata.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
+from typing import List
```
Comment (bkmartinjr) [inline tools/cell_census_builder/tests/anndata/test_anndata.py:1]: any reason to not include a simple test for the schema version?  (ie., that we drop any unsupported schema versions)?

Comment (bkmartinjr) [reply tools/cell_census_builder/tests/anndata/test_anndata.py:1]: I would also add a test to ensure it correctly handles the case where the AnnData lacks a raw (i.e., adata.raw is None)

Comment (ebezzi) [reply tools/cell_census_builder/tests/anndata/test_anndata.py:1]: Good idea, I will add both.

**Review by bkmartinjr — CHANGES_REQUESTED** [review]
No concerns with the tests, but recommend adding a test for the other conditions the code must handle:
* incorrect cellxgene schema version (drops the AnnData)
* no raw layer

File: tools/cell_census_builder/tests/anndata/test_anndata.py [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+
+    # The X matrix should have the same size as raw.X
+    assert h5ad.X.shape == h5ad.raw.X.shape
+
+    # The 4th column (correspnding to the added gene) is all zeros
+    added_col = h5ad.X.todense()[:, 3]
+    assert np.array_equal(added_col.A1, np.zeros(4))
+
+
+import pytest
```
Comment (Bento007) [inline tools/cell_census_builder/tests/anndata/test_anndata.py:66]: why is this import down here?

Comment (ebezzi) [reply tools/cell_census_builder/tests/anndata/test_anndata.py]: Was temporary. The fixtures have been moved to conftest.py

**Comment by ebezzi** [comment]
> No concerns with the tests, but recommend adding a test for the other conditions the code must handle:
> 
> * incorrect cellxgene schema version (drops the AnnData)
> * no raw layer

@bkmartinjr added a test for the incorrect schema version. Turns out the default datasets already have no raw layer, so that's already tested. I explicitly added a statement in the test description.

File: tools/cell_census_builder/anndata.py [hunk 3]
```diff
@@ -114,7 +114,7 @@ def _filter(ad: anndata.AnnData, retain_X: Optional[bool] = True) -> anndata.Ann
         if organism_ontology_term_id is not None:
             obs_mask = obs_mask & (ad.obs.organism_ontology_term_id == organism_ontology_term_id)
         if assay_ontology_term_ids is not None:
-            obs_mask = obs_mask & ad.obs.assay_ontology_term_id.isin(RNA_SEQ)
+            obs_mask = obs_mask & ad.obs.assay_ontology_term_id.isin(assay_ontology_term_ids)
```
Comment (ebezzi) [inline tools/cell_census_builder/anndata.py:117]: This is a bug in the current version of the code, as per discussion with @bkmartinjr yesterday.

**Comment by bkmartinjr** [comment]
> Turns out the default datasets already have no raw layer

Just want to confirm that we also test datasets _with_ a raw layer as well?  I believe you do, but confirming....

**Comment by ebezzi** [comment]
> > Turns out the default datasets already have no raw layer
> 
> Just want to confirm that we also test datasets _with_ a raw layer as well? I believe you do, but confirming....

Yes, that is tested by `test_open_anndata_equalizes_raw_and_normalized`. If you think it's worth it, I can add one test where the raw layer exists and has the same shape as the normalized one.

**Review by bkmartinjr — APPROVED** [review]
LGTM!  Thanks for the enhancements.

**Pull request merged**